Gems/PhysX/Code/Tests/CharacterGameplayControllerTests.cpp

```diff
@@ -137,18 +139,90 @@ namespace PhysX
+
+        basis.m_gameplayController->SetFallingVelocity(originalVelocity + expectedVelocity);
+        auto endVelocity = basis.m_gameplayController->GetFallingVelocity();
+
+        EXPECT_THAT(endVelocity.GetZ(), testing::FloatNear(expectedVelocity.GetZ(), 0.001f));
+    }
+
+    TEST_F(PhysXDefaultWorldTest, CharacterGameplayController_FallingVelocitySetsWhileMoving)
+    {
+        GameplayTestBasis basis(m_testSceneHandle, DefaultGravityMultiplier, DefaultGroundDetectionBoxHeight);
+
+        const AZ::Vector3 expectedVelocity(0.0f, 0.0f, 22.0f);
+        auto originalVelocity = basis.m_gameplayController->GetFallingVelocity();
+
+        for (int i = 0; i < 10; i++)
+        {
+            auto setVelocity = originalVelocity + expectedVelocity + AZ::Vector3(0.0f, 0.0f, float(i));
+            basis.m_gameplayController->SetFallingVelocity(setVelocity);
+            basis.Update();
+
+            auto endVelocity = basis.m_gameplayController->GetFallingVelocity();
+            const auto gravity = basis.m_testScene->GetGravity();
+
+            // The expected velocity should be decelerated by gt when g is gravity acceleration and t is the length of time
+            const float expectedDeceleration = gravity.GetZ() * basis.m_timeStep;
+
+            EXPECT_THAT(endVelocity.GetZ(), testing::FloatNear(setVelocity.GetZ() + expectedDeceleration, 0.001f));
+        }
+    }
+
+    TEST_F(PhysXDefaultWorldTest, CharacterGameplayController_SetGroundDetectionHeight)
+    {
+        GameplayTestBasis basis(m_testSceneHandle, DefaultGravityMultiplier, DefaultGroundDetectionBoxHeight);
+
+        const auto originalHeight = basis.m_gameplayController->GetGroundDetectionBoxHeight();
+        const float expectedHeight = 10.0f;
+
+        basis.m_gameplayController->SetGroundDetectionBoxHeight(expectedHeight + originalHeight);
+        auto endHeight = basis.m_gameplayController->GetGroundDetectionBoxHeight();
+
+        EXPECT_THAT(endHeight, testing::FloatNear(expectedHeight + originalHeight, 0.001f));
+    }
+
+    TEST_F(PhysXDefaultWorldTest, CharacterGameplayController_GroundDetectedWhileMoving)
+    {
+        GameplayTestBasis basis(m_testSceneHandle, DefaultGravityMultiplier, DefaultGroundDetectionBoxHeight);
+        basis.AddGround();
+
+        const AZ::Transform startingPosition = AZ::Transform::CreateTranslation(AZ::Vector3(0.0f, 0.0f, 0.1f));
+
+        basis.m_controller->SetTransform(startingPosition);        
+        const auto startingDetected = basis.m_gameplayController->IsOnGround();
+
+        bool groundDetected = false;
+
+        for (int i = 0; i < 10; i++)
+        {
+            basis.Update();
+
+            if (basis.m_gameplayController->IsOnGround())
+            {
+                groundDetected = true;
+                break;
+            }
+        }
+
+        EXPECT_THAT(groundDetected, testing::AllOf(testing::IsTrue(), testing::Ne(startingDetected)));
```

Comment:
suggestion: Could we add some comments for this test, it's a little tricky to see exactly what's going on. Do we need the `testing::IsTrue` here anymore? (maybe that was from some testing?)

Reply:
I'll update the comments for all the tests with this PR.

This is needed. The condition is making sure groundDetected is actually true while also checking that the starting state was not detected in the first place.  If for some reason the test got into a state where the starting detection was True and the end detection is False we would throw a false passing situation. 

This is the logical breakdown:

```
if (groundDetected) && (groundDetected != startingDetected):
     Pass
else:
    Fail
```

Reply:
I see understood, thank you for clarifying! I see `groundDetected` is being checked against all conditions in the `testing::AllOf` matcher (apologies I think I initially read it as equivalent to `if (true)`, but I see what it's doing now). Thanks!

Reply:
You could just write

EXPECT_FALSE(startingDetected);
EXPECT_TRUE(groundDetected);

which I would say is much easier to read.

And you could just move the first expectation to line 224 and say

EXPECT_FALSE(basis.m_gameplayController->IsOnGround());

Reply:
After discussion with @greerdv, I've adjusted the logic here to follow his advice.